plugins/janus_sip.c

```diff
@@ -2671,6 +2707,18 @@ static void *janus_sip_handler(void *data) {
 				g_hash_table_insert(masters, GUINT_TO_POINTER(session->master_id), session);
 				janus_mutex_unlock(&sessions_mutex);
 			}
+
+			json_t *header_prefixes_json = json_object_get(root, "incoming_header_prefixes");
+			if(header_prefixes_json && json_is_array(header_prefixes_json)) {
+				size_t index;
+				json_t *value;
+
+				json_array_foreach(header_prefixes_json, index, value) {
+					const char *header_prefix = g_strdup(json_string_value(value));
```

Comment:
No validation is done on the JSON. First of all, it looks like `incoming_header_prefixes` is not in `register_parameters` (see how we do for `headers`), which means the plugin will not automatically reject it if it's not an array. Besides, you're assuming the content of that array will be a string (`json_string_value`), which can't be given for granted.